File: 169-majority-element/majority-element.java
 class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int n=nums.length/2;
        
        for(int word : nums){
            map.put(word,map.getOrDefault(word,0)+1);
        }
        for(int word : nums){
            if(map.get(word)>n){
                return word;
            }
        }
    return -1;
    }
 }  

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        seens = {}
        n = len(nums)
        for i in range(0,n):
            if nums[i] in seens:
                seens[nums[i]] += 1
            else:
                seens[nums[i]] = 1
            if seens[nums[i]] > n // 2:
                return nums[i]
        